rpcserver.go

```diff
@@ -2482,12 +2525,22 @@ func (r *rpcServer) sendPayment(stream *paymentStream) error {
 					PaymentRoute:    marshalledRouted,
 				})
 				if err != nil {
-					errChan <- err
+					rpcsLog.Errorf("Failed sending "+
+						"response: %v", err)
+
+					select {
+					case errChan <- err:
+					default:
+					}
 					return
 				}
 			}()
 		}
 	}
+
+	// Wait for all goroutines to finish before closing the stream.
+	wg.Wait()
```

Comment:
Not meaning to gravedig or anything, but I was looking at this code today.  Won't this fail to be called in some circumstances and therefore not wait for all of the goroutines to finish?  Like if one of the payments failed to send and sent an error on `errChan`, then this send loop would call `return err` and therefore never call this.